interceptors/logging/interceptors.go

```diff
@@ -50,64 +52,117 @@ func (c *reporter) PostCall(err error, duration time.Duration) {
 	logger.With(c.opts.durationFieldFunc(duration)...).Log(c.opts.levelFunc(code), fmt.Sprintf("finished %s %s call", c.kind, c.typ))
 }
 
-func (c *reporter) PostMsgSend(interface{}, error, time.Duration) {}
+// PostMsgSend logs the details of the servingObject that is flowing out of the rpc.
+// resp object wrt server.
+// Log the details of the request, skip if the object is response.
+func (c *reporter) PostMsgSend(resp interface{}, err error, duration time.Duration) {
+	if !c.opts.shouldLog(interceptors.FullMethod(c.service, c.method), err) {
+		return
+	}
+	// If the first message is logged skip the rest of the logging.
+	// If the serving object is response, skip the logging.
+	if c.firstMessageLogged || c.isServer {
+		return
+	}
+	c.firstMessageLogged = true
 
-func (c *reporter) PostMsgReceive(interface{}, error, time.Duration) {}
+	logger := c.logger.With(extractFields(tags.Extract(c.ctx))...)
+	logger = logger.With("grpc.recv.duration", duration.String())
+	if err != nil {
+		logger = logger.With("error", fmt.Sprintf("%v", err))
+	}
+
+	logger.Log("logged the start of the request.", fmt.Sprintf("request object: %v", resp))
+
```

Comment:
We do same thing twice, can we move to function?

Reply:
I have made modifications, so this suggestion is outdated :stuck_out_tongue: